design-2.md

```diff
@@ -0,0 +1,119 @@
+# Design-2
+
+A simple but bad program for the empty room would be as follows:
+
+    action go right
+    can move right {
+        move right
+        go right
+    }
+    action go up
+    can move up {
+        move up
+        go up
+    }
+
+    action go left
+    can move left {
+        move left
+        go left
+    }
+    action go right 2
+    can move right {
+        move right
+        go right 2
+    }
+    move down
+    go left
```

Comment:
Adam touches on a lot of good points. I think the program with abstractions below clarifies a lot of these points. However, it seems a bit complicated for a simple task.